### DISCUSSION TITLE
Best practices

### DISCUSSION THREAD

**Discussion by Veercodeprog** [ParanoidUser/yolo#646] [Ask Away]
i am a newbie in mern,i can develop in mern as well
i am looking for a resource that follows best practices for folder structure and authentications?
i prefer resources that does not uses redux library

**Comment by FleetAdmiralJakob** [comment]
Do you really want to host auth and user management by yourself? If not I would recommend [Clerk](https://clerk.com/). Otherwise here is a good example for a mern stack: https://github.com/flaviuse/mern-authentication . Since I'm not a fan of mern and using a fullstack framework such as next.js I would recommend for a self-hosted solution [Auth.js](https://authjs.dev/)

**Reply by Veercodeprog** [reply]
not going to use the m in mern.
i have experience with node.not much with react.
Not sure if next.js  needed in my app.
can you list some pointers how it helps my app.also whats clerk?

**Reply by FleetAdmiralJakob** [reply]
Clerk is a service that hosts all of your user's stuff. With the small library from them, you can simply add a sign in page with just adding one component. Also you can really easily get the data everywhere. I recommend you watching this video to learn more about it: https://www.youtube.com/watch?v=5hlROBrm1Yk&ab_channel=LiveCode247 .

**Reply by Veercodeprog** [reply]
i am afraid it does not give me freedom to style my ui,and manage states as i want

**Reply by FleetAdmiralJakob** [reply]
Nope, it let's you style your components. Watch this video: https://www.youtube.com/watch?v=aQN4bJ1yc-k&ab_channel=Joshtriedcoding , here he shows you that you can style everything before the lock 🔒

**Reply by FleetAdmiralJakob** [reply]
And if you rly don't want this you can try Auth.js or copy the method from this repo: https://github.com/flaviuse/mern-authentication
